src/datasets/arrow_dataset.py

```diff
@@ -2468,8 +2475,6 @@ def init_buffer_and_writer():
                                 writer.write(example)
                 else:
                     for i in pbar:
-                        if drop_last_batch and i + batch_size > input_dataset.num_rows:
-                            continue
```

Comment:
I think there's no test for the `drop_last_batch` parameter. Maybe let's add one ?